File: src/main/kotlin/org/tera201/vcsmanager/util/PathUtils.java
package org.tera201.vcsmanager.util;

public class PathUtils {

    public static String fullPath(String path) {
        if(path==null) return null;

        if(path.startsWith("~"))
            path = path.replaceFirst("^~",System.getProperty("user.home"));

        return path;
    }
}

File: src/main/kotlin/org/tera201/vcsmanager/util/PathUtils.kt
package org.tera201.vcsmanager.util

object PathUtils {
    fun fullPath(path: String?): String? {
        var path = path ?: return null

        if (path.startsWith("~")) path = path.replaceFirst("^~".toRegex(), System.getProperty("user.home"))

        return path
    }
}
